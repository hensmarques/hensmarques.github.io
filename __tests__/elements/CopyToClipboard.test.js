import React from 'react';
import CopyToClipboard from '../../src/elements/CopyToClipboard';

describe('<CopyToClipboard/>', () => {
  it('renders with default props', () => {
    const defaultProps = {
      text: '',
      length: null,
    };
    shallow(<CopyToClipboard {...defaultProps} />);
  });

  const props = {
    text: '7s89dfs7-dfjsd893r-dfsf93',
    length: 25,
  };

  it('renders with custom props and matches snapshot', () => {
    const wrapper = shallow(<CopyToClipboard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('copies text when clicked', () => {
    const wrapper = shallow(<CopyToClipboard {...props} />);
    const spyExecCommand = jest.fn();
    Object.defineProperty(global.document, 'execCommand', { value: spyExecCommand });
    wrapper.find('.element-copy-text').simulate('click');
    expect(spyExecCommand).toHaveBeenCalledTimes(1);
    expect(spyExecCommand).toHaveBeenCalledWith('copy');
  });
});
