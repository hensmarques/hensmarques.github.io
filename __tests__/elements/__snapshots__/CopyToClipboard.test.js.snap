// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CopyToClipboard/> renders with custom props and matches snapshot 1`] = `
<span
  className="copy-text"
>
  <span
    className="element-copy-text"
    data-html={true}
    data-tip="Click to copy"
    onClick={[Function]}
  >
    7s89dfs7-dfjsd893r-dfsf93
  </span>
  <ReactTooltip
    clickable={false}
    insecure={true}
    resizeHide={true}
    wrapper="div"
  />
</span>
`;
